compiler/include/preNormalizeOptimizations.h

```diff
@@ -18,14 +18,14 @@
  * limitations under the License.
  */
 
-#ifndef _AUTO_LOCAL_ACCESS_H_
-#define _AUTO_LOCAL_ACCESS_H_
+#ifndef _PRE_NORMALIZE_OPTIMIZATIONS_H_
+#define _PRE_NORMALIZE_OPTIMIZATIONS_H_
 
 #include "CallExpr.h"
 #include "symbol.h"
 
 // interface for normalize
-void autoLocalAccess();
+void doPreNormalizeOptimizations();
```

Comment:
Minor, but I don't find the name particularly helpful. Maybe `preNormalizeArrayOptimizations`?

I think it's worth calling out that it's pre-normalize since source level semantics are important for these optimizations, but wonder if it'd be helpful to specify a little more.